frontend/src/modules/Worksheets/components/WorksheetTextToSQLEditor.js

```diff
@@ -0,0 +1,241 @@
+                label="Team"
+                name="team"
+                value={worksheet ? worksheet.SamlAdminGroupName : ''}
+                variant="outlined"
+              />
+            </Box>
+            <Box sx={{ p: 2 }}>
+              <TextField
+                fullWidth
+                label="Database"
+                name="database"
+                onChange={(event) => {
+                  handleDatabaseChanges(event);
+                }}
+                select
+                value={selectedDatabase}
+                variant="outlined"
+                InputProps={{
+                  endAdornment: (
+                    <>
+                      {loadingDatabases ? (
+                        <CircularProgress color="inherit" size={20} />
+                      ) : null}
+                    </>
+                  )
+                }}
+              >
+                {databaseOptions.length > 0 ? (
+                  databaseOptions.map((database) => (
+                    <MenuItem key={database.datasetUri} value={database}>
+                      {database.label}
+                    </MenuItem>
+                  ))
+                ) : (
+                  <MenuItem disabled>No databases found</MenuItem>
+                )}
+              </TextField>
+            </Box>
+
+            <Box sx={{ p: 2 }}>
+              <Autocomplete
+                multiple
+                options={tableOptions.map((t) => t.GlueTableName)}
+                value={selectedTables}
+                onChange={(_, newValue) => handleTablesChange(newValue)}
+                renderInput={(params) => (
+                  <>
+                    {loadingTables ? (
+                      <CircularProgress color="inherit" size={20} />
+                    ) : (
+                      <TextField
+                        {...params}
+                        variant="outlined"
+                        label="Select Tables"
+                        placeholder="Tables"
+                      />
+                    )}
+                  </>
+                )}
+                renderTags={(value, getTagProps) =>
+                  value.map((option, index) => (
+                    <Chip
+                      variant="outlined"
+                      label={option}
+                      {...getTagProps({ index })}
+                    />
+                  ))
+                }
+                disabled={!selectedDatabase}
+                fullWidth
+                margin="normal"
+              />
+            </Box>
+            <Box sx={{ p: 2 }}>
+              <TextField
+                fullWidth
+                label="Prompt"
+                multiline
+                rows={4}
+                value={prompt}
+                onChange={(e) => handlePromptChange(e.target.value)}
+                variant="outlined"
+              />
+            </Box>
+            <Box sx={{ p: 2 }}>
+              <LoadingButton
```

Comment:
nit: disable button if it is going to throw an error. e.g. if environment, database and tables are not selected, disable it

Reply:
good call out - will do